gamemodes/terrortown/gamemode/client/cl_eventpopup.lua

```diff
@@ -1,53 +1,150 @@
 ---
 -- @class EPOP
 -- @author Mineotopia
--- @desc A event popup system that works alongside the MSTACK system to display important messages
+-- @desc An event popup system that works alongside the MSTACK system to display important messages
 
-EPOP = {}
+local TIMER_IDENTIFIER = "epop_removal_timer"
+
+local defaultMessage = {
+	title = {
+		text = "A Test Popup, now with a multiline title, how NICE."
+	},
+	subtitle = {
+		text = "Well, hello there! This is a fancy popup with some special information. The text can be also multiline, how fancy! Ugh, I could add so much more text if I'd had any ideas..."
+	},
+	iconTable = {},
+	time = CurTime() + 5
+}
+
+EPOP = EPOP or {}
+
+EPOP.messageQueue = EPOP.messageQueue or {}
 
 ---
 -- Adds a popup message to the @{EPOP}
 -- @param string|table title The title of the popup that will be displayed in large letters (can be a table with `text` and `color` attribute)
--- @param string|table text An optional description that will be displayed below the title (can be a table with `text` and `color` attribute)
--- @param number disp_time The render duration of the popup
--- @param table icon_tbl An optional set of icon materials that will be rendered below the popup
+-- @param [opt]string|table subtitle An optional description that will be displayed below the title (can be a table with `text` and `color` attribute)
+-- @param [default=4]number displayTime The render duration of the popup
+-- @param [opt]table iconTable An optional set of icon materials that will be rendered below the popup
+-- @param [default=true]boolean blocking If this is false, this message gets instantly replaced if a new message is added
+-- @return string Returns a unique id generated for this message
 -- @realm client
-function EPOP:AddMessage(title, text, disp_time, icon_tbl)
-	self.msg = {
-		title = {},
-		text = {}
-	}
+function EPOP:AddMessage(title, subtitle, displayTime, iconTable, blocking)
+	local id = "epop_unique_id_" .. tostring(CurTime() * 10000)
```

Comment:
I don't really like this way to create an unique identifier. Wouldn't it be better to just use a global variable, which is increased every time AddMessage is called?

Reply:
and the benefit of this would be... ?

Reply:
To be certain about unique identifiers.

Reply:
I mean, this time is in 1/1000 seconds resolution

Reply:
Sure, in practive, this won't probably ever result in a collision, but it is still bad style to assign an identifier in that way.

Reply:
okay, I'll change it then